flink-runtime/src/test/java/org/apache/flink/runtime/state/VoidNamespaceSerializerUpgradeTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state;
+
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.api.common.typeutils.TypeSerializerMatchers;
+import org.apache.flink.api.common.typeutils.TypeSerializerSchemaCompatibility;
+import org.apache.flink.api.common.typeutils.TypeSerializerUpgradeTestBase;
+import org.apache.flink.testutils.migration.MigrationVersion;
+
+import org.hamcrest.Matcher;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import static org.hamcrest.Matchers.is;
+
+/**
+ * A {@link TypeSerializerUpgradeTestBase} for {@link VoidNamespaceSerializer}.
+ */
+@RunWith(Parameterized.class)
+public class VoidNamespaceSerializerUpgradeTest extends TypeSerializerUpgradeTestBase<VoidNamespace, VoidNamespace> {
+
+	private static final String SPEC_NAME = "void-namespace-serializer";
+
+	public VoidNamespaceSerializerUpgradeTest(TestSpecification<VoidNamespace, VoidNamespace> testSpecification) {
+		super(testSpecification);
+	}
+
+	@Parameterized.Parameters(name = "Test Specification = {0}")
+	public static Collection<TestSpecification<?, ?>> testSpecifications() throws Exception {
+
+		ArrayList<TestSpecification<?, ?>> testSpecifications = new ArrayList<>();
+		for (MigrationVersion migrationVersion : migrationVersions) {
+			testSpecifications.add(
+				new TestSpecification<>(
+					SPEC_NAME,
+					migrationVersion,
+					VoidNamespaceSerializerSetup.class,
+					VoidNamespaceSerializerVerifier.class));
+		}
+		return testSpecifications;
+	}
+
+	// ----------------------------------------------------------------------------------------------
+	//  Specification for "voidnamespace-serializer"
+	// ----------------------------------------------------------------------------------------------
+
+	/**
```

Comment:
Does our checkstyle configuration cover tests? I don't think I like these comments very much because they don't add any information, they just repeat the class name. They are public only to make the `ClassRelocator` that the test base uses work.

Reply:
I thought that our checkstyle configuration cover test class, because if I did not add the comments, will encounter the error such as below when validating
```
[ERROR] src/test/java/org/apache/flink/streaming/api/operators/TimerSerializerUpgradeTest.java:[84] (javadoc) JavadocType: Missing a Javadoc comment.
```
and this is the reason why I added this [hotfix commit](https://github.com/aljoscha/flink/commit/62d979f47254138f95210aef74ee86df665979a9). 

But when I double-checked the PojoXXXSetup class today, they do not have the java doc comments and do not violate the checkstyle. I did not figure out the reason currently, will continue to figure out the reason.

Reply:
It could be that we have checkstyle enabled for tests only in some modules

Reply:
Yes, `PojoXXSetup` classes locate in `flink-core` module, and this module uses a custom suppress file ([suppressions-core.xml](https://github.com/aljoscha/flink/blob/master/tools/maven/suppressions-core.xml)) for checkstyle which skip the `JavadocType` check.
Do you think we need to add a custom suppress file for every touched module, or add a  meaningful java doc for every test class?

Reply:
Maybe we can add the same javadoc to all of them. Something like: `This class is only public to work with the ClassRelocator`.

Reply:
got it, will update the javadoc.